src/borges_spec.erl

```diff
@@ -63,13 +68,36 @@ make_internal_specification_structure(Module) ->
     StorageName = Module:name(),
     Subsets = Module:subsets(),
     MainStorage = Module:main_storage(),
-    SubsetsMap = maps:from_list([{Name, Def} || #{name := Name} = Def <- Subsets]),
+%    SubsetsMap = maps:from_list([{Name, Def} || #{name := Name} = Def <- Subsets]),
+    SubsetsMap = maps:from_list(make_internal_subset_structure(Subsets)),
+
     #{name => StorageName,
       main_storage => MainStorage,
       handler_pid => undefined,
       subsets => SubsetsMap,
       module => Module}.
 
+make_internal_subset_structure([]) ->
+    [];
+make_internal_subset_structure([#{name := Name} = Subset0 | Rest]) ->
+    Extend = maps:get(extend, Subset0, fun default_extend/2),
+    Reduce = maps:get(reduce, Subset0, fun default_reduce/2),
+    Subset = Subset0#{
+        extend => Extend,
+        reduce => Reduce
+    },
+    [{Name, Subset} | make_internal_subset_structure(Rest)].
+
+default_extend(not_found, Data) ->
+    sets:from_list([Data]);
```

Comment:
sets:add_element(Data, sets:new()); is cheaper/cleaner?

Reply:
this also adds an option to use otp24 versioning if we need to use set representation..
sets:new([{version, 2}])

Reply:
I find the current version cleaner. What is cheaper is just speculation.
Regarding sets v2, that'd add a hard dependency on otp 24 for the library. I don't think it is worth it